gyft.py

```diff
@@ -111,38 +64,146 @@ def merge_slots(in_dict):
+            subject_timings[timetable_dict[day][time][0]].append([flattened_time, timetable_dict[day][time][2]])
+        subject_timings = merge_slots(subject_timings)
+        for time in list(timetable_dict[day].keys()):
+            flattened_time = int(time[:time.find(':')])
+            if (flattened_time < 6):
+                flattened_time += 12
+            if (not flattened_time == subject_timings[timetable_dict[day][time][0]][0]):
+                del (timetable_dict[day][time])
+            else:
+                timetable_dict[day][time][2] = subject_timings[timetable_dict[day][time][0]][1]
+
+    # get course names
+    r = s.post(COURSES_URL, headers=headers, data=coursepage_details)
+    sub_dict = {item["subno"]: item["subname"] for item in r.json()}
+    sub_dict = {k: v.replace("&amp;", "&") for k, v in sub_dict.items()} # replacing &amp; with &
+
+    # add course name to dict
+    for day in timetable_dict.keys():
+        for time in timetable_dict[day]:
+            course_code = timetable_dict[day][time][0]
+            timetable_dict[day][time].append(sub_dict[course_code])
+    
+    return timetable_dict
+
+def main():
+    overwrite = True
+    generate = True
+
+    args = parse_args()
+
+    if args.del_events:
+        delete_calendar()
+        gen = input("\nWould you like to generate a new timetable? (y/n): ")
+        if gen.lower() == 'n':
+            generate = False
+    
+    if generate:
+        INPUT_FILENAME = args.input if args.input else "data.txt"
+        OUTPUT_FILENAME = args.output if args.output else "timetable.ics"
+
+        # check if data.txt exists
+        if osp.exists(INPUT_FILENAME):
+            ow = input(f"Timetable file {INPUT_FILENAME} exists. Do you want to overwrite it? (y/n): ")
+            print()
+            if ow.lower() == 'n':
+                overwrite = False
+
+        if overwrite:
+            s = requests.Session()
+            _, ssoToken = erp.login(headers, s)
+            print()
+
+            if SEM_BEGIN.month > 6:
+                # autumn semester
+                SEM_NO = (int(SEM_BEGIN.strftime("%y"))-int(erp.ROLL_NUMBER[:2]))*2 + 1
+            else:
+                # spring semester
+                SEM_NO = (int(SEM_BEGIN.strftime("%y"))-int(erp.ROLL_NUMBER[:2])) + 2
+
+            timetable_details = {
+                "ssoToken": ssoToken,
+                "module_id": '16',
+                "menu_id": '40',
+            }
+
+            coursepage_details = {
+                "ssoToken": ssoToken,
+                "semno": SEM_NO,
+                "rollno": erp.ROLL_NUMBER,
+                "order": "asc"
+            }
+
+            timetable_dict = scrape_timetable(s, timetable_details, coursepage_details)
+
+            with open(INPUT_FILENAME, 'w') as outfile:
+                json.dump(timetable_dict, outfile, indent = 4, ensure_ascii=False)
+            
+            print(f"\nTimetable saved to {INPUT_FILENAME} file.")
```

Comment:
```suggestion
            print(f"Timetable saved to {INPUT_FILENAME} file.\n")
```
<img width="400" alt="image" src="https://github.com/metakgp/gyft/assets/86282911/15ffae35-731d-4971-bc35-ae43aaf08d02">

Fixes two lines above and no lines below